main.go

```diff
@@ -38,30 +37,28 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	reporter, err := output.GetReporter(config.Reports)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	plan := plan.New(config)
 
-	fmt.Printf("Waiting on WAIT_FOR=%v\n\n", plan.Config.WaitForHosts)
+	if err = reporter.Start(plan); err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("\nWaiting on WAIT_FOR=%v\n\n", plan.Config.WaitForHosts)
 	execute.Wait(plan.Config.WaitForHosts, time.Duration(30)*time.Second)
 
 	fmt.Printf("\nExecuting Matrix...\n\n")
 	results := execute.Run(plan)
 
-	fail := func(err error) {
-		os.Exit(1)
-	}
-
-	reporter, err := output.GetReporter(config.Reports)
-	if err != nil {
-		fail(err)
-	}
-
-	if err = reporter.Start(config); err != nil {
-		fail(err)
-	}
 	for test := range results {
 		reporter.Next(test)
 	}
 	if err := reporter.End(); err != nil {
-		fail(err)
+		log.Fatal(err)
```

Comment:
ok, so here, reporter.End() should not fail. It it fails, its an error in crossdock, do we really want to hide the error here?

Or is it in the contract of reporters, that if a reporter function returns an error, it means "I have printed errors shit, now please exit". Because in this case, I would argue that this is not clear at all, and we should return (status, err). status being an enum of what action to perform (continue or exit).

Any example why a reporting function would want to abort the whole crossdock by purpose?